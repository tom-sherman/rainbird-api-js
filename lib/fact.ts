export type RainbirdObject = string | number | boolean

export class Fact {
  subject: string
  relationship: string
  object: RainbirdObject
  cf: number

  constructor({ subject, relationship, object, cf = 100 }: FactOptions) {
    if (cf > 100) {
      throw new Error('cf must be between 1-100')
    }

    this.subject = subject
    this.relationship = relationship
    this.object = object
    this.cf = cf
  }
}

interface FactOptions {
  subject: string
  relationship: string
  object: RainbirdObject
  cf?: number
}
