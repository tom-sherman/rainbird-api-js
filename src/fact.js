const rb = require('./rainbird')

class Fact {
  constructor ({ apiKey, apiDomain, sessionId, factID, ...fact }) {
    this.data = fact
    this.id = factID
    this.apiKey = apiKey
    this.apiDomain = apiDomain || Fact.DEFAULT_API_DOMAIN
    this.sessionId = sessionId
  }

  async audit () {
    return rb.call({
      method: 'GET',
      path: `/analysis/evidence/${this.id}`,
      apiKey: this.apiKey,
      apiDomain: this.apiDomain
    })
  }

  static get DEFAULT_API_DOMAIN () {
    return rb.API_DOMAIN
  }
}

module.exports = Fact
