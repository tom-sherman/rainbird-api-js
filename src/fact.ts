export type RainbirdObject = string | number | boolean

export class Fact {
  subject: string
  relationship: string
  object: RainbirdObject
  cf: number

  constructor({ subject, relationship, object, cf = 100 }: Fact) {
    this.subject = subject
    this.relationship = relationship
    this.object = object
    this.cf = cf
  }
}
