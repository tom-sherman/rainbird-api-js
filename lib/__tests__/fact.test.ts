import { Fact } from '../fact'

describe('fact', () => {
  test('should construct without cf', () => {
    expect(
      new Fact({ subject: 's', relationship: 'r', object: 'o' })
    ).toMatchSnapshot()
  })

  test('can set cf to below 100', () => {
    expect(
      new Fact({ subject: 's', relationship: 'r', object: 'o', cf: 80 })
    ).toMatchSnapshot()
  })

  test('should throw when setting cf out of bounds of 1-00', () => {
    ;[
      () => new Fact({ subject: 's', relationship: 'r', object: 'o', cf: 101 }),
      () => new Fact({ subject: 's', relationship: 'r', object: 'o', cf: 0 }),
      () => new Fact({ subject: 's', relationship: 'r', object: 'o', cf: -1 })
    ].forEach(errorFn => expect(errorFn).toThrow('cf must be between 1-100'))
  })

  test('should throw when setting cf to a non number', () => {})
})
