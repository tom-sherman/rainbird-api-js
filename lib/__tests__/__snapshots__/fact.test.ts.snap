// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fact can set cf to below 100 1`] = `
Fact {
  "cf": 80,
  "object": "o",
  "relationship": "r",
  "subject": "s",
}
`;

exports[`fact should construct without cf 1`] = `
Fact {
  "cf": 100,
  "object": "o",
  "relationship": "r",
  "subject": "s",
}
`;
